### PR TITLE
records: addition of ``literature`` and ``authors`` marshmallow schemas 

### PR DESCRIPTION
**Pull request by drjova**

### CODE REVIEW

**Comment by drjova** [comment]
**DO NOT MERGE**

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            kekscan_link = self.get_link_for_kekscan_schema(value)
+            if kekscan_link is None:
+                return missing
+            return kekscan_link
+        return "{}{}".format(url_prefix, value)
+
+    def get_link_for_kekscan_schema(self, external_system_id_value):
+        extid = external_system_id_value.replace("-", "")
+        if (
+            len(extid) == 7
```
Comment (ammirate) [inline inspirehep/records/marshmallow/literature/common/external_system_identifier.py]: This is another place where would be nice to have some docstrings...Where this 7 come from, for example?

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        if schema == "kekscan":
+            kekscan_link = self.get_link_for_kekscan_schema(value)
+            if kekscan_link is None:
+                return missing
+            return kekscan_link
+        return "{}{}".format(url_prefix, value)
+
+    def get_link_for_kekscan_schema(self, external_system_id_value):
+        extid = external_system_id_value.replace("-", "")
+        if (
```
Comment (ammirate) [inline inspirehep/records/marshmallow/literature/common/external_system_identifier.py]: parenthesys? never seen in Python :P

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            if not self.is_missing_url_name_or_link(extid)
+        ]
+
+    def is_missing_url_name_or_link(self, external_system_id):
+        return (
+            external_system_id.get("url_link") is None
+            or external_system_id.get("url_name") is None
+        )
+
+    def take_first_id_foreach_url_name(self, external_system_ids):
```
Comment (ammirate) [inline inspirehep/records/marshmallow/literature/common/external_system_identifier.py]: I would preferer `get_first_id_foreach_url_name` :P

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        taken_url_names = set()
+        unique_ids = []
+        for external_system_id in external_system_ids:
```
Comment (ammirate) [inline inspirehep/records/marshmallow/literature/common/external_system_identifier.py]:
```python
unique_ids = set()
for external_system_id in external_system_ids:
    url_name = external_system_id.get("url_name")
    unique_ids.add(url_name)
return unique_ids
```

File: run-tests-ci.sh [hunk 5]
```diff
@@ -11,7 +11,10 @@
 set -e
 
 if [ "$SUITE" == "unit" ]; then
+<<<<<<< HEAD
```
Comment (ammirate) [inline run-tests-ci.sh]: ops :eyes:

File: run-tests-ci.sh [hunk 6]
```diff
@@ -11,7 +11,10 @@
...
     pipenv run isort -rc -c -df && \
+=======
+>>>>>>> tests: addition of tests for marshmallow custom fields
```
Comment (ammirate) [inline run-tests-ci.sh]: end ops :eyes:

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py (hunk 2 above)
Comment (drjova) [reply inspirehep/records/marshmallow/literature/common/external_system_identifier.py]: black :P

**Comment by drjova** [comment]
@ammirate updated, I didn't do everything since I didn't refactor the code

File: inspirehep/records/marshmallow/literature/common/author.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    first_name = fields.Method("get_first_name", default=missing)
+    last_name = fields.Method("get_last_name", default=missing)
+
+    def get_first_name(self, data):
+        names = data.get("full_name", "").split(",", 1)
+
+        if len(names) > 1:
+            return names[1].replace(",", "").strip()
+
+        return names[0] or missing
```
[the remark this answers is not in the document]
Comment (drjova) [reply inspirehep/records/marshmallow/literature/common/author.py]: actually it's should be ok, it's ugly but will not touch it for now
```python
names = ['']

names[0] == ''

'' or missing
```

File: inspirehep/records/marshmallow/literature/common/external_system_identifier.py (hunk 4 above)
Comment (drjova) [reply inspirehep/records/marshmallow/literature/common/external_system_identifier.py]: the tests are failing with this change, I'll leave as is for now

**Comment by drjova** [comment]
@ammirate a second 👀 ?

**Pull request merged**